Client/loader/CInstallManager.cpp

```diff
@@ -527,6 +542,81 @@ void MigrateFile(const SString& strFilenameOld, const SString& strFilenameNew)
     }
 }
 
+//////////////////////////////////////////////////////////
+//
+// CInstallManager::_ProcessGtaVersionCheck
+//
+// Make sure gta exe is the correct version
+//
+//////////////////////////////////////////////////////////
+SString CInstallManager::_ProcessGtaVersionCheck()
+{
+    SString strGtaExe = PathJoin(GetGTAPath(), "gta_sa.exe");
+    SString strPatchBase = PathJoin(GetGTAPath(), "audio/CONFIG/PakFiles.dat");
+    SString strPatchDiff = PathJoin(GetMTASAPath(), "mta/data/gta_sa_diff.dat");
+    SString strGtaExeMd5 = "170B3A9108687B26DA2D8901C6948A18";
+    SString strPatchBaseMd5 = "DB1E657A3BAAFBB86CD1B715C5282C66";
+    // Reset any bytes MTA may have changed in gta_sa.exe
+    std::vector<CFileGenerator::SResetItem> gtaExeResetList = {
+        {0x0000008B, 1, {0x42}},                              // bTimestamp
+        {0x00000096, 1, {0x0F}},                              // bLargeMem
+        {0x000000DF, 1, {0x00}},                              // bDep
+        {0x000000F8, 8, {0x00}},                              // bNvightmare ExportDir
+        {0x004A1AD0, 96, {0x00}},                             // bNvightmare ExportTable
+        {0x004C4588, 4, {0x01, 0x00, 0x00, 0x00}},            // bNvightmare ExportValue
+    };
+    CFileGenerator fileGenerator(strGtaExe, strGtaExeMd5, gtaExeResetList, strPatchBase, strPatchDiff);
+
+    // Need to fix gta_sa.exe?
+    if (!fileGenerator.IsGenerationRequired())
+    {
+        return "ok";
+    }
+
+    // Check required GTA file is correct
+    if (GenerateHashHexStringFromFile(EHashFunctionType::MD5, strPatchBase) != strPatchBaseMd5)
+    {
+        AddReportLog(5053, SString("_ProcessGtaVersionCheck: Incorrect file '%s'", *strPatchBase));
+        SString strMessage(_("MTA:SA cannot continue because the following files are incorrect:"));
+        strMessage += "\n\n" + strPatchBase;
+        BrowseToSolution("gengta_pakfiles", ASK_GO_ONLINE, strMessage);
```

Comment:
I'm guessing https://updatesa.mtasa.com/sa/trouble/?tr=gengta_pakfiles will be created when this is merged?

Reply:
Correct